### PR TITLE
Add app name field and populate fields with first option

### PR DESCRIPTION
**Pull request by carabasdaniel**
This pull request should also fix duplification issue

### CODE REVIEW

File: src/CloudFoundry.VisualStudio/ProjectPush/PublishProfileEditorResources.cs [hunk 1]
```diff
@@ -232,6 +232,13 @@ private void EnterRefresh()
 
         private void ExitRefresh()
         {
+            if (string.IsNullOrWhiteSpace(this.PublishProfile.Organization))
```
Comment (viovanov) [inline src/CloudFoundry.VisualStudio/ProjectPush/PublishProfileEditorResources.cs]: This doesn't look ok.
Should not pick default every time refresh is exited. 
This code should be in [RefreshClient](https://github.com/cloudfoundry-incubator/cf-vs-extension/blob/develop/src/CloudFoundry.VisualStudio/ProjectPush/PublishProfileEditorResources.cs#L309)

File: src/CloudFoundry.VisualStudio/ProjectPush/PublishProfileEditorResources.cs [hunk 2]
```diff
@@ -358,6 +358,14 @@ public void Refresh(PublishProfileRefreshTarget refreshTarget)
             await this.RefreshSharedDomains();
             await this.RefreshPrivateDomains();
             await this.RefreshServiceInstances();
+
+            if (string.IsNullOrWhiteSpace(this.PublishProfile.Organization))
```
Comment (mihaibuzgau) [inline src/CloudFoundry.VisualStudio/ProjectPush/PublishProfileEditorResources.cs]: why is this not in the RefreshOrganization method ?

File: src/CloudFoundry.VisualStudio/ProjectPush/PublishProfileEditorResources.cs [hunk 3]
```diff
@@ -358,6 +358,14 @@ public void Refresh(PublishProfileRefreshTarget refreshTarget)
...
+            {
+                if (this.Orgs.Count > 0)
+                {
+                    this.PublishProfile.Organization = this.Orgs.FirstOrDefault().Name;
```
Comment (mihaibuzgau) [inline src/CloudFoundry.VisualStudio/ProjectPush/PublishProfileEditorResources.cs]: shouldn't this be done OnUIThread ?

File: src/CloudFoundry.VisualStudio/ProjectPush/PublishProfileEditorResources.cs [hunk 4]
```diff
@@ -403,21 +419,36 @@ public void Refresh(PublishProfileRefreshTarget refreshTarget)
...
+            OnUIThread(() =>
+            {
+                this.spaces.Clear();
+                foreach (var space in spacesList)
+                {
+                    this.spaces.Add(space);
+                }
+            });
+
+            if (string.IsNullOrWhiteSpace(this.PublishProfile.Space))
```
Comment (mihaibuzgau) [inline src/CloudFoundry.VisualStudio/ProjectPush/PublishProfileEditorResources.cs]: can this be moved on the previous OnUIThread ?

File: src/CloudFoundry.VisualStudio/ProjectPush/PublishProfileEditorResources.cs [hunk 5]
```diff
@@ -460,18 +502,32 @@ public void Refresh(PublishProfileRefreshTarget refreshTarget)
...
+            OnUIThread(() => 
+            { 
+                this.stacks.Clear();
+                foreach (var stack in stacksList)
+                {
+                    this.stacks.Add(stack);
+                }
+            });
+
+            if (string.IsNullOrWhiteSpace(this.PublishProfile.Application.StackName))
```
Comment (mihaibuzgau) [inline src/CloudFoundry.VisualStudio/ProjectPush/PublishProfileEditorResources.cs]: can this be moved on the previous OnUIThread?

**Comment by mihaibuzgau** [comment]
:+1:

**Pull request merged**